contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/querydsl/sort/CollectionFilterSortHalFormsPayloadMetadataContributor.java

```diff
@@ -0,0 +1,208 @@
+import org.springframework.hateoas.mediatype.MediaTypeConfigurationCustomizer;
+import org.springframework.hateoas.mediatype.MessageResolver;
+import org.springframework.hateoas.mediatype.hal.forms.HalFormsConfiguration;
+import org.springframework.hateoas.mediatype.hal.forms.HalFormsOptions;
+import org.springframework.hateoas.mediatype.html.HtmlInputType;
+
+@RequiredArgsConstructor
+public class CollectionFilterSortHalFormsPayloadMetadataContributor extends
+        SortHandlerMethodArgumentResolverSupport implements HalFormsPayloadMetadataContributor,
+        MediaTypeConfigurationCustomizer<HalFormsConfiguration> {
+
+    private final CollectionFiltersMapping collectionFiltersMapping;
+    private final Repositories repositories;
+    private final MessageResolver messageResolver;
+
+    @Override
+    public Stream<PropertyMetadata> contributeToCreateForm(Class<?> domainType) {
+        return Stream.empty();
+    }
+
+    @Override
+    public Stream<PropertyMetadata> contributeToUpdateForm(Class<?> domainType) {
+        return Stream.empty();
+    }
+
+    @Override
+    public Stream<PropertyMetadata> contributeToSearchForm(Class<?> domainType) {
+        var sortOptions = collectionFiltersMapping.forDomainType(domainType)
+                .forSorting()
+                .documented()
+                .filters()
+                .flatMap(filter -> {
+                    return Stream.of(
+                            Direction.ASC,
+                            Direction.DESC
+                    ).map(dir -> new SortOption(
+                            filter.getFilterName(),
+                            dir,
+                            messageResolver.resolve(new DirectionMessageSourceResolvable(domainType, filter, dir)),
+                            foldIntoExpressions(Sort.by(dir, filter.getFilterName()))
+                    ));
+                });
+        return Stream.of(new SortPropertyMetadata(sortOptions.toList()));
+    }
+
+    @Override
+    public HalFormsConfiguration customize(HalFormsConfiguration configuration) {
+        var sortParamName = getSortParameter(null);
+        for (Class<?> domainType : repositories) {
+            configuration = configuration.withOptions(domainType, sortParamName, pm -> {
+                if (pm instanceof SortPropertyMetadata sortPropertyMetadata) {
+                    return HalFormsOptions.inline(sortPropertyMetadata.getSortOptions())
+                            .withMinItems(0L)
+                            .withValueField("value")
+                            .withPromptField("prompt");
+                }
+
+                // Returning null here will just not add any options to this field.
+                // That is the expected fallback when the sortParam does turn out not to actually be SortPropertyMetadata,
+                // since that means the sort parameter was actually an existing field name on the domain type
+                return null;
```

Comment:
Is this covered in tests?

Reply:
It is covered now in a new test that has an entity with a `sort` property (which is not searchable)